workitem/workitem_repository.go

```diff
@@ -283,3 +285,25 @@ func (r *GormWorkItemRepository) List(ctx context.Context, criteria criteria.Exp
 
 	return res, count, nil
 }
+
+// GetCountsPerIteration fetches WI count from DB and returns a map of iterationID->WICountsPerIteration
+// This function executes following query to fetch 'closed' and 'total' counts of the WI for each iteration in given spaceID
+//		select iterations.id as IterationId, count(*) as Total,
+//			count( case fields->>'system.state' when 'closed' then 'A' else null end ) as Closed
+//			from work_items, iterations
+//			where iterations.space_id = `space ID` and iterations.id = fields->>'system.iteration'
+//			group by IterationId
+func (r *GormWorkItemRepository) GetCountsPerIteration(ctx context.Context, spaceID uuid.UUID) (map[string]WICountsPerIteration, error) {
+	var res []WICountsPerIteration
+	db := r.db.Table("work_items").Select(`iterations.id as IterationId, count(*) as Total,
+				count( case fields->>'system.state' when 'closed' then '1' else null end ) as Closed`).Joins(`left join iterations 
+					on iterations.id::text = fields->>'system.iteration'`).Where(`iterations.space_id = ? and work_items.deleted_at IS NULL`, spaceID).Group(`IterationId`).Scan(&res)
```

Comment:
done in https://github.com/almighty/almighty-core/pull/706/commits/100586221a359244046fbc3c4d2ea8535e5e7ad1